megamek/src/megamek/client/commands/ShowTileCommand.java

```diff
@@ -36,12 +52,22 @@ public String run(String[] args) {
         try {
             int i = 3;
             String str = "", report = "";
-            Coords coord = new Coords(Integer.parseInt(args[1]) - 1, Integer
-                                                                             .parseInt(args[2]) - 1);
+            Coords coord;
```

Comment:
Spitballing here to reduce the cognitive load of all the parsing and do-while logic:
```java
    /**
     * Reads a list of coordinates from the command arguments.
     * 
     * The acceptable argument formats are:
     *   - XX, YY
     *       A specific coordinate.
     *   - XX, YY, DIR0, DIR1, ..., DIRN
     *       One or more directions, beginning from the specific hex and
     *       walking along the path specified by the directions.
     *   - DIR0, DIR1, ..., DIRN
     *       One or more directions, beginning from the current hex and
     *       walking along the path specified by the directions.
     * 
     * @param args An array of arguments containing an optional starting
     *             coordinate, followed by zero or more directions.
     * @return A list of {@link Coords} to show, or an empty list if no
     *         valid arguments were found.
     */
    private List<Coords> readCoords(String[] args) {
        if (args.length == 0) {
            return Collections.emptyList();
        }

        List<Coords> coords = new ArrayList<>();

        Coords coord = getClient().getCurrentHex();
        for (int ii = 0; ii < args.length; ++ii) {
            String arg = args[ii].toUpperCase();
            if (directions.contains(arg)) {
                coords.add(coord.translated(arg));
            } else if (ii == 0 && args.length >= 2) {
                coord = new Coords(Integer.parseInt(args[0]), Integer.parseInt(args[1]));
                coords.add(coord);
            }
        }

        return coords;
    }
```

Used like:
```java
for (Coords coord : readCoords(args)) {
   // the logic within the do loop
}
```

Reply:
This may need a special case to ensure the current hex is added if its not a specific hex, but the sentiment holds.